### PR TITLE
Added support for line continuations.

### LINKED ISSUE #2473: Add support for continuations in properties.
```kotlin
    @Suppress("WrongPropertyKeyValueDelimiter", "TrailingSpacesInProperty")
    @Issue("https://github.com/openrewrite/rewrite/issues/2473")
    @Test
    fun escapedNewLines() {
        val source = """
                \
                
                k\
                 e\
                
                  y \
                   = \
                
                    v\
                
                     al\
                      ue\
                       
            """.trimIndent()
        val props = PropertiesParser().parse(source)[0] as Properties.File

        assertThat(props.content.map { it as Properties.Entry }.map { it.key })
            .hasSize(1)
            .containsExactly("key")
        assertThat(props.content.map { it as Properties.Entry }.map { it.value.text })
            .hasSize(1)
            .containsExactly("value")

        val outputCapture = PrintOutputCapture(InMemoryExecutionContext())
        val printer = PropertiesPrinter<InMemoryExecutionContext>()
        printer.visitFile(props, outputCapture)
        assertThat(outputCapture.out.toString()).isEqualTo(source)
    }
```

### PR DESCRIPTION
**Pull request by traceyyoshima**
fixes #2473

### CODE REVIEW

File: rewrite-properties/src/main/java/org/openrewrite/properties/PropertiesParser.java [hunk 1]
```diff
@@ -76,23 +75,34 @@ private Properties.File parseFromInput(Path sourceFile, EncodingDetectingInputSt
...
         for (char c : s.toCharArray()) {
-            if (isEscapedNewLine) {
+            if (inContinuation) {
                 if (Character.isWhitespace(c)) {
+                    continuation.append(c);
                     continue;
                 } else {
-                    isEscapedNewLine = false;
+                    continuationPositions.put(buff.length(), continuation.toString());
+                    continuation.delete(0, continuation.length());
```
Comment (tkvangorder) [inline rewrite-properties/src/main/java/org/openrewrite/properties/PropertiesParser.java:92]:
```suggestion
                    continuation.setLength(0);
```

File: rewrite-properties/src/main/java/org/openrewrite/properties/PropertiesParser.java [hunk 2]
```diff
@@ -76,23 +75,34 @@ private Properties.File parseFromInput(Path sourceFile, EncodingDetectingInputSt
...
+                    inContinuation = true;
+
+                    // Move the escape character to the continuation.
                     buff.deleteCharAt(buff.length() - 1);
+                    continuation.append("\\");
+
+                    continuation.append(c);
                 } else {
                     Properties.Content content = extractContent(buff.toString(), prefix);
                     if (content != null) {
```
Comment (tkvangorder) [inline rewrite-properties/src/main/java/org/openrewrite/properties/PropertiesParser.java:108]:
```suggestion
                    if (content != null) {
                        if (!continuationPositions.isEmpty() && content instanceof Properties.Entry) {
                            content = content.withMarkers(content.getMarkers().addIfAbsent(new EntryContinuation(randomId(), continuationPositions)));
                            continuationPositions = new HashMap<>();
                        }
```

File: rewrite-properties/src/main/java/org/openrewrite/properties/PropertiesParser.java [hunk 3]
```diff
@@ -107,8 +117,16 @@ private Properties.File parseFromInput(Path sourceFile, EncodingDetectingInputSt
             prev = c;
         }
 
+        if (inContinuation) {
+            continuationPositions.put(buff.length(), continuation.toString());
+            continuation.delete(0, continuation.length());
```
Comment (tkvangorder) [inline rewrite-properties/src/main/java/org/openrewrite/properties/PropertiesParser.java:122]:
```suggestion
            continuation.setLength(0);
```

**Review by tkvangorder — CHANGES_REQUESTED** [review]
See my changes. We need to attach the continuations to each entry (and reset the continuations each time) 

Without these changes, you can see the side effects by adding a second property to your test: 

```
                \
                
                k\
                 e\
                
                  y \
                   = \
                
                    v\
                
                     al\
                      ue
  
                fred=fred
```

**Comment by tkvangorder** [comment]
Also, if we mutate an entry that has continuation markers, we either need to clear the markers or compute updates to the positioning. Failing to do that, we are going to run into issues with printing.

**Comment by traceyyoshima** [comment]
> Also, if we mutate an entry that has continuation markers, we either need to clear the markers or compute updates to the positioning. Failing to do that, we are going to run into issues with printing.

The issue is fixed via updates to with(s) and a few changes to the ProperiesParser.

**Comment by sambsnyd** [comment]
I'm going to close this PR for now since it isn't ready to commit and no one is actively working on this

**Pull request closed without merging**